saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/jackson2/AbstractMixinTests.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.saml2.jackson2;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+
+import org.springframework.security.jackson2.SecurityJackson2Modules;
+import org.springframework.security.saml2.provider.service.authentication.DefaultSaml2AuthenticatedPrincipal;
+
+abstract class AbstractMixinTests {
```

Comment:
Instead of relying on inheritance here, I think this could be a static factory class that each of the tests invokes. For example, there is `TestRelyingPartyRegistrations` that constructs different kinds of relying party registration configurations. This could instead be `TestSaml2JsonPayloads`.

Reply:
Move payloads to factory class. Not sure if i also should move the creation of the ObjectMapper to a factory or still use the before method.

Reply:
I think the before method is fine, it's probably simple enough to repeat inside each test instead of inheriting.